File: programmers/LV.2/카펫.py
def is_decimal(n):
	a=int(n**0.5)
	for i in range(2,a+1):
		if n%i==0:
			return False
		else:
			continue
	return True


def solution(brown,yellow):
	if is_decimal(yellow):
		width=(yellow+2)
		height=3
		return [width,height]
	else:
		# 2로 나눴을 때 가능한 숫자만 높이로 지정한다. 
		max_height=(int(yellow//2))
		for h in range(1,max_height+1):
			if yellow%h==0:
				garo=int((yellow/h)+2)
				sero=h+2
				if garo*sero-yellow==brown:
					return [garo,sero]

print(solution(12,4))
		

File: 프로그래머스/2/42842. 카펫/카펫.java
// 문제 풀이 솔루션 
// 약수를 구해서 나누어지는 것이라면 ((가로+2) *2) + ((세로 +2) *2)-4 가 갈색의 수와 같다면, 정답임. 

class Solution {
    public int[] solution(int brown, int yellow) {
        int[] answer = new int[2] ;
        
        for(int i=1;i<=Math.sqrt(yellow);i++){
            if(yellow%i==0){
                int b = yellow/i; // 나머지
                if((b+2)*2+(i+2)*2-4 == brown){
                    if(i>=b){
                        answer[0]=i+2;
                        answer[1]=b+2;
                    }else{
                        answer[1]=i+2;
                        answer[0]=b+2;
                    }
                }
            }
        }
        
        
        return answer;
    }
}